api/app.go

```diff
@@ -459,6 +460,19 @@ func updateApp(w http.ResponseWriter, r *http.Request, t auth.Token) (err error)
 			return permission.ErrUnauthorized
 		}
 	}
+	if updateData.Pool != "" {
+		currentPool, err := pool.GetPoolByName(a.GetPool())
```

Comment:
Linter failed here with 
```
api/app.go:464::warning: declaration of "err" shadows declaration at api/app.go:398 (vetshadow)
```
To avoid this you can declare `var currentPool, newPool *provision.Pool` in the beginning of the if block, and use `=` instead of `:=`

Reply:
@cezarsa Could you please explain to me why this happens?

Reply:
Variables in Go have block scope, by using `:=` inside the if block you're actually declaring a new variable `err` that only happens to have the same name as the external `err` variable. This can cause subtle and hard to find errors as changing the value of the inner variable doesn't change the outer one. That's the reason why go vet complains about it. Check https://play.golang.org/p/JD-at0fAbg for a better example.

Reply:
Thank you very much. I wasn't aware of the block scope.